java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java

```diff
@@ -130,7 +130,7 @@ private void handleSuppressWarning(List<AnnotationTree> annotationTrees, Tree tr
     List<String> rules = Lists.newArrayList();
     for (AnnotationTree annotationTree : annotationTrees) {
       if (isSuppressWarningsAnnotation(annotationTree)) {
-        startLine = ((JavaTree) annotationTree).getLine();
+        startLine = startLineIncludingTrivia(tree);
```

Comment:
The start line computation was starting from the annotation, now you are starting from the tree, which is not the same.
Moreover, I have the feeling that it won't work if there is comment in-between annotations (see code below), probably worth to add the case.

```
class A {

  /**
   * TODO
   */
  @MyAnnotation
  /**
   * TODO
   */
  @SuppressWarnings("repo:S1135")
  void foo() {

  }
}

@interface MyAnnotation {
}
```